pkg/storage/kube/build.go

```diff
@@ -34,7 +40,48 @@ func (s *store) GetBuild(id string) (*brigade.Build, error) {
 	return b, err
 }
 
-// Get creates a new project and writes it to storage.
+// DeleteBuild deletes a build.
+func (s *store) DeleteBuild(bid string, options storage.DeleteBuildOptions) error {
+	opts := meta.ListOptions{
+		LabelSelector: fmt.Sprintf(jobFilter, bid),
+	}
+	delOpts := meta.NewDeleteOptions(0)
+	pods, err := s.client.CoreV1().Pods(s.namespace).List(opts)
+	if err != nil {
+		return err
+	}
+	if options.SkipRunningBuilds {
```

Comment:
I'm wondering if these lines 53-67 can be consolidated into one pod iteration loop?  Something like
```
for _, p := range pods.Items {
  if p.Labels["component"] == "build" {
    if options.SkipRunningBuilds && (p.Status.Phase == v1.PodRunning || p.Status.Phase == v1.PodPending) {
      return errors.New("skipping because build is still running")
    } else {
      log.Printf("Deleting pod %q", p.Name)
      if err := s.client.CoreV1().Pods(s.namespace).Delete(p.Name, delOpts); err != nil {
          log.Printf("failed to delete job pod %s (continuing): %s", p.Name, err)
      }
    }
  }
}
```

Or, if the above isn't quite right, just looking for some way to condense logic/iterations.  If it just doesn't work this way, no worries!

Reply:
Created https://github.com/Azure/brigade/issues/696 to track